### PR TITLE
WebGPU: Split out webgpu struct info

### PR DESCRIPTION
**Pull request by lokokung**
Splitting this out allows for easier maintainability.

### CODE REVIEW

File: src/struct_info_webgpu.json [hunk 1]
```diff
@@ -0,0 +1,536 @@
+[
+    // ===========================================
+    // WebGPU
+    //   NOTE: This section is auto-generated.
```
Comment (sbc100) [inline src/struct_info_webgpu.json:4]: Maybe update the comment here?  "This file is ..."

File: system/include/webgpu/README.md [hunk 2]
```diff
@@ -15,14 +15,13 @@ is included here because it is strongly tied to an exact `webgpu.h` revision.
...
-    - `system/lib/webgpu/webgpu_cpp.cpp`
-1. Paste the contents of [`library_webgpu_enum_tables.js`](https://source.chromium.org/chromium/chromium/src/+/main:out/linux-Debug/gen/third_party/dawn/emscripten-bits/library_webgpu_enum_tables.js) over the "Map from enum number to enum string" section of [`library_webgpu.js`](../../../src/library_webgpu.js)
-1. Paste [`webgpu_struct_info.json`](https://source.chromium.org/chromium/chromium/src/+/main:out/linux-Debug/gen/third_party/dawn/emscripten-bits/webgpu_struct_info.json) over the "WebGPU" section of [`struct_info.json`](../../../src/struct_info.json).
+1. Paste the contents of [`library_webgpu_enum_tables.js`](https://source.chromium.org/chromium/chromium/src/+/main:out/linux-Debug/gen/third_party/dawn/src/emdawnwebgpu/library_webgpu_enum_tables.js) over the "Map from enum number to enum string" section of [`library_webgpu.js`](../../../src/library_webgpu.js)
+1. Paste [`struct_info_webgpu.json`](https://source.chromium.org/chromium/chromium/src/+/main:out/linux-Debug/gen/third_party/dawn/src/emdawnwebgpu/struct_info_webgpu.json) into [`struct_info_webgpu.json`](../../../src/struct_info_webgpu.json).
```
Comment (sbc100) [inline system/include/webgpu/README.md:24]: Perhaps just say "Copy .. to .. " here now?

**Review by sbc100 — APPROVED** [review]
Nice!

File: src/struct_info_webgpu.json [hunk 3]
```diff
@@ -0,0 +1,536 @@
...
+                "layout",
+                "vertex",
+                "primitive",
+                "depthStencil",
+                "multisample",
+                "fragment"
+            ]
+        }
+    }
+]
```
Comment (sbc100) [inline src/struct_info_webgpu.json:536]: Missing newline at EOF

File: src/struct_info_webgpu.json (hunk 1 above)
Comment (lokokung) [reply src/struct_info_webgpu.json]: Done.

File: system/include/webgpu/README.md (hunk 2 above)
Comment (lokokung) [reply system/include/webgpu/README.md]: Done.

File: src/struct_info_webgpu.json (hunk 3 above)
Comment (lokokung) [reply src/struct_info_webgpu.json]: Done.

**Review by sbc100 — COMMENTED** [review]
lgtm.. but I'll wait for @kainino0x for final approval

**Review by kainino0x — APPROVED** [review]
LGTM, didn't realize we could do this! Makes updating much easier (even though we won't be doing rolling actively into Emscripten for a little while)

**Pull request merged**